source/interact-data/transaction.txt

```diff
@@ -0,0 +1,324 @@
+.. _mongoid-data-txn:
+
+=========================
+Transactions and Sessions
+=========================
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: code example, ACID compliance, multi-document, ruby odm
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to use {+odm+} to perform
+**transactions**. Transactions allow you to perform a series of operations
+that change data only if the entire transaction is committed.
+If any operation in the transaction does not succeed, the driver stops the
+transaction and discards all data changes before they ever become
+visible. This feature is called :wikipedia:`atomicity <Atomicity_(database_systems)>`.
+
+In MongoDB, transactions run within logical **sessions**. A
+session is a grouping of related read or write operations that you
+want to run sequentially. Sessions enable causal consistency for a group
+of operations, which means that all processes in your application agree on
+the order of causally-related operations.
+
+Sessions allow you to run operations in an **ACID-compliant**
+transaction that meets an expectation of atomicity, consistency,
+isolation, and durability. MongoDB guarantees that the data involved in
+your transaction operations remains consistent, even if the operations
+encounter unexpected errors.
+
+In {+odm+}, you can perform transactions by using either of the
+following APIs:
+
+- :ref:`mongoid-txn-convenient`: {+odm+} manages the life cycle of the
+  transaction. You can use this API in {+odm+} v9.0 and later.
+
+- :ref:`mongoid-txn-core`: You must manage the life cycle of the
+  transaction. You can use this API in {+odm+} v6.4 and later.
+
+The :ref:`mongoid-txn-session` section describes how to make changes to
+your data from within a session without performing a transaction.
+
+.. _mongoid-txn-convenient:
+
+Convenient Transaction API
```

Comment:
The term "Convenient Transaction API" refers to  the [drivers API](https://github.com/mongodb/specifications/blob/master/source/transactions-convenient-api/transactions-convenient-api.md). It can be used in Mongoid like this (but this it probably not the best idea):

```ruby
person.with_session do |session|
  session.with_transaction do
    person.save!
  end
end
```

Maybe we should not using this term to avoid confusion.

Reply:
Reverted the terminology to use high-level and low-level